### PR TITLE
#124 Quick Search

### LINKED ISSUE #124: Quick search using Elasticsearch
Use Elasticsearch to provide a quick search on the main page. It will search through:
- Entries
- Caves
- Grotto (groups)

### PR DESCRIPTION
**Pull request by GodefroiRoussel**
Quick search in front end with icons, highlights and searchbar.
In the API the route /api/v1/search.
The feature can be tested in local thanks to the localDeploy script that will install and create containers for Elasticsearch and Logstash. 

Fix #124

### CODE REVIEW

File: assets/js/react/helpers/Entity.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+export const entityOptionForSelector = (option) => {
+  const highlights = [];
+  Object.keys(option.highlights).forEach((key) => {
+    highlights.push({ [key]: option.highlights[key].join(' [...] ') });
+  });
+
+  return (
+    <React.Fragment>
+      {option.type === 'entry' ? <EntityIcon src="../../../../images/entry.svg" alt="Entry icon" /> : ''}
```
Comment (maximenathan) [inline assets/js/react/helpers/Entity.jsx]: C'est pas mieux de mettre le path absolut ?

Comment (Clm-Roig) [reply assets/js/react/helpers/Entity.jsx]: Oui, je confirme, le fix par Godefroi arrive dans un instant :wink:

File: config/datastores.js [hunk 2]
```diff
@@ -31,9 +31,10 @@ module.exports.datastores = {
    *                                                                          *
    ***************************************************************************/
 
+  //TODO: Change the url to 'mysql://sailsuser:grottocepassword@mysqlgrotto/grottoce' for dockerization
   default: {
     adapter: require('sails-mysql'),
-    url: 'mysql://root:root@localhost/grottoce'
+    url: 'mysql://sailsuser:grottocepassword@localhost:33060/grottoce',
```
Comment (Kurdty) [inline config/datastores.js]: This file must not be commited

**Comment by Clm-Roig** [comment]
Reopening the branch to recheck if everything is ok from scratch (launch of the logstash and elasticsearch via docker mainly).

**Pull request merged**